### PR TITLE
[10.0][MIG] base_kanban_stage

### PR DESCRIPTION
**Pull request by lasley**
This is a forward port of #679 

* Rename manifest
* Change openerp references to odoo
* Bump version

### CODE REVIEW

**Comment by lasley** [comment]
Added missing commits from #679 - this is ready for review

**Comment by dreispt** [comment]
Hmm, your tests are failing...

**Comment by lasley** [comment]
And now they aren't failing! Good for review.

**Review by dreispt — APPROVED** [review]

**Comment by lasley** [comment]
cc @laslabs

**Comment by jcdrubay** [comment]
I wanted to test but it seems not easy to test because this module is abstract and therefore meaningless unless being used.

I tried to test combining the server-tool branch from LasLabs from this Pull Request + the `LasLabs/vertical-medical:release/10.0` which contains an example for medical_prescription but I am getting an error...probably due to my environment :/

Is there any easier way to test?

Anyway, I believe we should move this PR forward quickly to help @lasley merging the LasLabs:vertical-medical to OCA.

The code looks good and includes with some tests :+1:

**Comment by lasley** [comment]
@jcdrubay - Yeah I wish there was a better way to test this functionally, but yeah your way is pretty much the best. 

Our incubator v10 has it in the oca_dependencies, and the `medical_prescription_stage` is already merged using it, so it can be tested directly on v10 branch of our Runbot if you'd like - https://runbot.laslabs.io/runbot/repo/git-github-com-laslabs-vertical-medical-git-41

I identified in https://github.com/LasLabs/vertical-medical/pull/119#issuecomment-275571183 that there seems to be a bug that if a record is not in the stage, the stage will not actually appear on the view after a refresh. 

This can be replicated by going to Medical => Dispensing => Add New Column => A Name => Add => Refresh the Page => New magic trick - Disappearing Stages

cc @obulkin

**Comment by lasley** [comment]
https://github.com/OCA/server-tools/pull/681#issuecomment-275572678 is fixed in f7a3904 by using [`group_expand`](https://github.com/odoo/odoo/blob/879c323578e157ce8f8a93a4d3703d71d102afed/odoo/models.py#L1664) & updating to new API

File: base_kanban_stage/tests/test_base_kanban_abstract.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        """ It builds a model from model_cls in order to test abstract models.
+
+        Note that this does not actually create a table in the database, so
+        there may be some unidentified edge cases.
+
+        Args:
+            model_cls (openerp.models.BaseModel): Class of model to initialize
+        Returns:
+            model_cls: Instance
+        """
```
Comment (obulkin) [inline base_kanban_stage/tests/test_base_kanban_abstract.py:27]: There are some issues with this docblock:

- As currently written, the method does actually create a table for the model. That happens as part of `_auto_init()`
- The var name in the `Returns` section is wrong

Comment (lasley) [reply base_kanban_stage/tests/test_base_kanban_abstract.py:27]: This came from template. Added note to PR https://github.com/OCA/maintainer-tools/pull/242/files#r98305473

**Review by jcdrubay — APPROVED** [review]
I tested in runbot of Laslabs for prescription and it works well.
I had reviewed the code already.
:+1:

File: base_kanban_stage/models/base_kanban_abstract.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+class BaseKanbanAbstract(models.AbstractModel):
+    """ Inherit from this class to add support for Kanban stages to your model.
+    All public properties are preceded with kanban_ in order to isolate from
+    child models, with the exception of stage_id, which is a required field in
+    the Kanban widget and must be defined as such. """
+
+    _name = 'base.kanban.abstract'
+    _order = 'kanban_priority desc, kanban_sequence'
+    _group_by_full = {
```
Comment (lasley) [inline base_kanban_stage/models/base_kanban_abstract.py:16]: This can be removed

**Comment by dreispt** [comment]
Looks like it's good to merge... doing it.
